_posts/2017-12-28-Secure-your-architecture-part2.md

```diff
@@ -0,0 +1,211 @@
+First step is installing the Java Cryptography Extension on our local pc.
+>JCE provides a framework and implementation for encryption, key generation, key agreement and message authentication code algorithms. 
+Downloads are available for Java 6, 7 and 8.
+This will allow the config server to use the encryption tool of the JCE.
+
+After the installation, the next step will be securing the config server by adding Spring security to the classpath and configuring your Basic/OAuth2 authentication.
+{% highlight maven %}
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-security</artifactId>
+</dependency>
+{% endhighlight %}
+
+
+## Key Management
+The config server supports encryption and decryption with a symmetric key or an asymmetric key-pair.
+The choice of which key you will need is within your security terms. 
+The symmetric key is the easiest way to set up but less secure than the asymmetric one. 
+To set up a symmetric key, you just assign a string to the key holder. 
+`encrypt.key=<key>`
+
+To configure these asymmetric keys, we will need a keystore created by the keytool utility from the JDK.
+The public key will encrypt and the private key will decrypt your data.
+
+To create a keystore you can do something like this in your commandline:
+
+{% highlight bash %}
+$> keytool -genkeypair -alias <keyname> -keyalg RSA -keysize 4096 -sigalg SHA512withRSA \
+  -dname "CN=Config Server,OU=JWorks,O=Ordina,L=Mechelen,S=State,C=BE" \
+  -keypass <secret> -keystore config-server.jks -storepass <password>
+{% endhighlight %}
+
+This will generate a keystore for the config server to use. 
+Place it in your repository project and configure it in your yml.
+
+Example yml in the config server:
+
+{% highlight yml %}
+encrypt:
+  key-store:
+    location: classpath:config-server.jks // resource location
+    password: <password> // to unlock the keystore
+    alias: config-server-key // to identify which key in the store is to be used
+    secret: <secret>
+{% endhighlight %}
+
+## Encryption
+To encrypt the data, start up your config server locally and enter this in your command line.
+
+{% highlight bash %}
+$> export SECRET=$(curl -X POST --data-urlencode secret http://user:password@localhost:<port>/encrypt)
+$> echo "datasource.password=$SECRET" >> application-dev.yml
+{% endhighlight %}
+
+When the encryption is done, we get a encrypted piece of data in your configuration in the form of:
+{% highlight yml %}
+spring:
+  datasource:
+    username: dbuser
+    password: '{cipher}FKSAJDFGYOS8F7GLHAKERGFHLSAJ'
+{% endhighlight %}
+
+## What to store where?
+When designing your config server, you have different options on where to store and fetch your sensitive data. 
+
+### Using a Git repository
+The default and most common way most of us use it is via private git repositories where we store our sensitive data where the config server can fetch it.
```

Comment:
... us use is via ... (drop the "it")
Git

Reply:
I would also change the "where the config server can fetch it" part to "and to which our config server has access" to make it more readable.